### ISSUE TITLE
Error running NVDA with python 3: 'louis' not found

### ISSUE THREAD

**Issue by destroyersboy** [nvaccess/nvda#10264]
### Steps to reproduce:
1. NVDA is cloned with Git:
`git clone --recurse https: // github.com/nvaccess/nvda`

2. Switch to NVDA:
`cd nvda`

3. NVDA is run with python3:
`py -3-32 source/nvda.pyw` or `cd source`, `py -3-32 nvda.pyw`
### Actual behavior:
1. NVDA plays start sound, windows 10 error sound is heard in parallel.
2. No response can be received from NVDA, nvda is installed on the system.
### Expected behavior:
1. NVDA should run.
2. If any steps are taken incorrectly, they should be included in NVDA's source run documentation.
#### NVDA installed/portable/running from source:
from source
#### NVDA version:
Sunday, September 22, 2019 **GitHub repository**
#### Windows version:
OS Name: Microsoft Windows 10 Pro
Version: 10.0.18358 Build 18358

#### Name and version of other software in use when reproducing the issue:
``py -3-32 --version
Python 3.7.3``

#### NVDA's log:
```
INFO - __main__ (23:17:56.043):
Starting NVDA version source-master-0e4bb4a
INFO - core.main (23:17:56.308):
Config dir: C:\Users\Destroyerpc\Documents\nvda\source\userConfig
INFO - config.ConfigManager._loadConfig (23:17:56.308):
Loading config: .\userConfig\nvda.ini
INFO - core.main (23:17:56.527):
Using Windows version 10.0.18358 workstation
INFO - core.main (23:17:56.527):
Using Python version 3.7.3 (v3.7.3:ef4ec6ed12, Mar 25 2019, 21:26:53) [MSC v.1916 32 bit (Intel)]
INFO - core.main (23:17:56.527):
Using comtypes version 1.1.7
INFO - core.main (23:17:56.527):
Using configobj version 5.1.0 with validate version 1.0.1
CRITICAL - __main__ (23:17:56.761):
core failure
Traceback (most recent call last):
  File "nvda.pyw", line 224, in <module>
    core.main()
  File "core.py", line 246, in main
    import appModuleHandler
  File "appModuleHandler.py", line 27, in <module>
    import NVDAHelper
  File "NVDAHelper.py", line 19, in <module>
    import eventHandler
  File "eventHandler.py", line 9, in <module>
    import api
  File "api.py", line 12, in <module>
    import review
  File "review.py", line 10, in <module>
    from NVDAObjects import NVDAObject, NVDAObjectTextInfo
  File "NVDAObjects\__init__.py", line 16, in <module>
    from displayModel import DisplayModelTextInfo
  File "displayModel.py", line 17, in <module>
    import mouseHandler
  File "mouseHandler.py", line 8, in <module>
    import gui
  File "gui\__init__.py", line 19, in <module>
    import ui
  File "ui.py", line 20, in <module>
    import braille
  File "braille.py", line 19, in <module>
    import louisHelper
  File "louisHelper.py", line 9, in <module>
    import louis
ModuleNotFoundError: No module named 'louis'
```

### Other questions
I searched Louis in the Include folder. I could not find it.
I installed and activated virtual venv.
Then `pip install louis`
Result: no run.
I copied the louis folder from NVDA's system folder to the Include folder, the liblouis subfolder.
Result: no run.

What should I do?

Yours

**Comment by lukaszgo1** [comment]
What about running scons source before attempting to start NVDA?

**Comment by josephsl** [comment]
Hi, did you build NVDA from source first? You need to do that before running NVDA from source. Thanks.

**Comment by destroyersboy** [comment]
@lukaszgo1 lukaszgo1, @josephsl josephsl

Excuse me, I'm new to NVDA.

VS 2019 installed on the machine.
Could this be related?
Scons went out here:
```
C:\Users\Destroyerpc\Documents\nvda>py -3-32 scons.py
scons: Reading SConscript files ...
scons: done reading SConscript files.
scons: Building targets ...
MIDL.EXE /nologo /win32 /c_ext /I build\x86 /tlb build\x86\iSimpleDOMNode.tlb /h build\x86\iSimpleDOMNode.h /iid build\x86\iSimpleDOMNode_i.c /proxy build\x86\iSimpleDOMNode_p.c /dlldata build\x86\iSimpleDOMNode_data.c build\x86\iSimpleDOMNode.idl
Processing build\x86\iSimpleDOMNode.idl
iSimpleDOMNode.idl
build\x86\iSimpleDOMNode.idl(102): warning C4067: unexpected tokens following preprocessor directive - expected a newline
build\x86\iSimpleDOMNode.idl(103): warning C4067: unexpected tokens following preprocessor directive - expected a newline
Processing C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um\objidl.idl
objidl.idl
Processing C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um\unknwn.idl
unknwn.idl
Processing C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared\wtypes.idl
wtypes.idl
Processing C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared\wtypesbase.idl
wtypesbase.idl
Processing C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared\basetsd.h
basetsd.h
Processing C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared\guiddef.h
guiddef.h
Processing C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um\oaidl.idl
oaidl.idl
Processing C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um\oaidl.acf
oaidl.acf
MIDL.EXE /I build\x86 /tlb build\x86\mathPlayerDLL.tlb /h build\x86\mathPlayerDLL.h /iid build\x86\mathPlayerDLL_i.c /proxy build\x86\mathPlayerDLL_p.c /dlldata build\x86\mathPlayerDLL_data.c build\x86\mathPlayerDLL.idl
Microsoft (R) 32b/64b MIDL Compiler Version 8.01.0622
Copyright (c) Microsoft Corporation. All rights reserved.
Processing build\x86\mathPlayerDLL.idl
mathPlayerDLL.idl
Processing C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um\oaidl.idl
oaidl.idl
Processing C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um\objidl.idl
objidl.idl
Processing C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um\unknwn.idl
unknwn.idl
Processing C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared\wtypes.idl
wtypes.idl
Processing C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared\wtypesbase.idl
wtypesbase.idl
Processing C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared\basetsd.h
basetsd.h
Processing C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared\guiddef.h
guiddef.h
Processing C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um\ocidl.idl
ocidl.idl
Processing C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um\oleidl.idl
oleidl.idl
Processing C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um\servprov.idl
servprov.idl
Processing C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um\urlmon.idl
urlmon.idl
Processing C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um\msxml.idl
msxml.idl
Processing C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um\oaidl.acf
oaidl.acf
Processing C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um\ocidl.acf
ocidl.acf
interfaceAction(["source\comInterfaces\SpeechLib.py", "source\comInterfaces\__pycache__\SpeechLib.cpython-37.pyc"], ["source\comInterfaces\__init__.py"])
scons: *** [source\comInterfaces\SpeechLib.py] Library not registered
scons: building terminated because of errors.

```

**Comment by Adriani90** [comment]
You must compile with vs2017

**Comment by destroyersboy** [comment]
> You must compile with vs2017

Isn't there a way to run NVDA with VS2019?
Are the components installed in VS2017 not included in VS2019?

I don't like VS very much. Because it takes up a lot of space, because of the obligation.
Is it not possible to download compiled resources required for NVDA?
So I get rid of the compile load.

**Comment by josephsl** [comment]
Hi, rest assured that someone (specifically, @LeonardDer) is looking into making your wish come true (in terms of Visual Studio 2019 requirement). CC @MichaelDCurran

**Comment by leonardder** [comment]
You can build nVDA with Visual Studio 2019 by installing the 15.1 (vs 2017) toolsets. Note that you need to do this for x86/x64 and arm64, and you also need to enable atl support for both.

**Comment by destroyersboy** [comment]
Hello again @leonardder ,

`C++ ATL for v141 build tools (ARM64), 7 of 52`
I think there are places I have to check here. I've never heard a label here like 15.1.
Which options do I have to check?

There should be a way to export our setup configuration in VS Installer, which could be more robust.

**Comment by leonardder** [comment]
Ugh, I"m sorry. I meant version 14.1 indeed, zo verison 141 is the right one to choose.

**Comment by destroyersboy** [comment]
Failed, all components related to ARM are checked. There's still a mistake coming from Scons.

I don't think a developer interested in NVDA needs to install VS2019/2017.

1. Really GB is too big. And VS spends much CPU.
2. The Python developer does not need to deal with issues such as C ++ and ARM.
3. Python offers a simple structure. It's not big and tiring.


It would not be right to discuss this here. Maybe we should create and discuss a different issue.

Discouraging.

**Comment by josephsl** [comment]
Hi, for NVDA, C++ and ARM support are required because of DLL’s in use. Let’s discuss this further on the NVDA development mailing list (nvda-devel) on Groups.IO. Thanks.

**Comment by leonardder** [comment]
Closing as per above comment.
